### ISSUE TITLE
Switch to typescript

### ISSUE THREAD

**Issue by dpriskorn** [diegodlh/zotero-cita#171]
I prefer strong typing to have better IDE experience and can lead to fewer errors in production because they are caught by the type system and validator.

**Comment by Dominic-DallOsto** [comment]
Yeah, this would be nice but I guess a lot of work. Is there some automated tool that would help with at least the first 90%?

Something like [this](https://www.npmjs.com/package/js-to-ts-converter) maybe.

**Comment by retorquere** [comment]
It should in principle be possible to just rename .js to .ts and set typescript to very lax checking. After that you could slowly ratchet up the checks. TS is supposed to be a strict superset of JS -- any valid JS is by stipulation valid TS.

Switching to typescript has been a huge boon for my plugins. The typechecking has saved my bacon many a time, and you get ESNext features to boot.

**Comment by jimmywarting** [comment]
[You might not need TypeScript... (syntax)](https://jimmywarting.github.io/you-might-not-need-typescript/) - you can have type safety without typescript syntax

**Comment by Dominic-DallOsto** [comment]
Work in progress [here](https://github.com/diegodlh/zotero-cita/tree/Dominic-DallOsto/issue171). Just ironing out a lot of small compiler errors mostly
